### PR TITLE
Added sudoku solver in javascript

### LINKED ISSUE #984: Sudoku solver in javascript
implementation of sudoku solver in javascript
please add the Gssoc label and assign this issue to me

### PR DESCRIPTION
**Pull request by mark-nakachon**
Fixes #984 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- I added Sudoku solver in javascript 

#### Languages Used:
- javascript

#### Files Added:
- Sudoku_Solver.js

### CODE REVIEW

**Comment by mark-nakachon** [comment]
please review my pr

**Comment by marwahmanbir** [comment]
Add a more descriptive title @mark-nakachon

**Comment by mark-nakachon** [comment]
Added indentation now you can merge it now

File: Sudoku_Solver/Sudoku_Solver.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
+const matrix = [
+    [5, 3, 0, 0, 7, 0, 0, 0, 0],
+	[6, 0, 0, 1, 9, 5, 0, 0, 0],
+	[0, 9, 8, 0, 0, 0, 0, 0, 0],
+	[8, 0, 0, 0, 6, 0, 0, 0, 3],
+	[4, 0, 0, 8, 0, 3, 0, 0, 1],
+	[7, 0, 0, 0, 2, 0, 0, 0, 6],
+	[0, 6, 0, 0, 0, 0, 2, 8, 0],
+	[0, 0, 0, 4, 1, 9, 0, 0, 5],
+	[0, 0, 0, 0, 8, 0, 0, 7, 9]
```
Comment (KanchanThareja) [inline Sudoku_Solver/Sudoku_Solver.js]: Indentation not fixed yet. Make sure you use 4 spaces instead of a tab.

Comment (marwahmanbir) [reply Sudoku_Solver/Sudoku_Solver.js]: @KanchanThareja it was already fixed. Check the latest commit.

Comment (KanchanThareja) [reply Sudoku_Solver/Sudoku_Solver.js]: @marwahmanbir yes it is my bad.

File: Sudoku_Solver/Sudoku_Solver.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    return false;
+};
+
+if (solve(matrix)) {
+    console.log(matrix);
+} else {
+    console.log(`sorry i can't solve`);
+}
+//output
+/* tested from leetcode https://leetcode.com/problems/sudoku-solver/#
```
Comment (marwahmanbir) [inline Sudoku_Solver/Sudoku_Solver.js]: Hey @mark-nakachon, is your code copied from https://leetcode.com/problems/sudoku-solver/?
@jainaman224 @gupta2140 @rishabhgarg25699

Comment (mark-nakachon) [reply Sudoku_Solver/Sudoku_Solver.js]: No, I just tested my code in that link whether it is passing all the test cases or not

**Comment by mark-nakachon** [comment]
I removed the link please merge this now

**Comment by GOVINDDIXIT** [comment]
@mark-nakachon Squash commits to one

**Comment by mark-nakachon** [comment]
@GOVINDDIXIT please merge this PR after merging this PR I will squash all my commits into one

**Comment by marwahmanbir** [comment]
@mark-nakachon the thing is that commits can only be squashed while the PR is open. You won't be able to squash them once this gets merged.

**Comment by jainaman224** [comment]
@KanchanThareja Please review.

**Comment by KanchanThareja** [comment]
@mark-nakachon you can connect with us on slack if you are facing any issue in squashing your commits.
@jainaman224  I approve the changes in this PR.

**Comment by jainaman224** [comment]
@mark-nakachon Please squash your commits.

**Comment by MastersAbh** [comment]
@mark-nakachon The PR is ready to merge. You can contact us if you want assistance in squashing of your commits.

**Comment by MastersAbh** [comment]
@mark-nakachon are you willing to go further with this PR ?

**Comment by MastersAbh** [comment]
@rishabhgarg25699 @jainaman224 have a look.

**Comment by rishabhgarg25699** [comment]
@mark-nakachon I think you are newbie in open source world . I am appreciating your work on this PR . But I am closing your PR As in your previous PR #958 , you also not squash commits .But we accept that PR but we not accept this PR .Try to learn squashing commit . We are very helpful . @jainaman224 If you are not with me then open this PR .

**Pull request closed without merging**